src/classes/HH_CampaignDedupeBTN_TEST.cls

```diff
@@ -1,124 +1,167 @@
-    
-    system.assert(con2.npo02__household__c != null);
-    insert con2;
-    
-    Campaign camp = new Campaign (
-      Name = 'MyCampaign',
-      Status = 'Planned',
-      isActive = true
-    );
-    insert camp;
-    
-    CampaignMember campm1 = new CampaignMember(
-      CampaignID = camp.id,
-      ContactID = con.id,
-      Status = 'Sent'      
-    );
-    insert campm1;
-    
-    CampaignMember campm2 = new CampaignMember(
-      CampaignID = camp.id,
-      ContactID = con2.id,
-      Status = 'Sent'       
-    );
-    insert campm2;
-        
-    //now run our report
-    Test.startTest();
-    ApexPages.StandardController sc = new ApexPages.StandardController(camp);
-    HH_CampaignDedupeBTN_CTRL deduper = new HH_CampaignDedupeBTN_CTRL(sc);
-    PageReference pr = deduper.runReport();
-    Test.stopTest();
-    
-    String campm1Status = [select Status from CampaignMember where id = :campm1.id].Status;
-    String campm2Status = [select Status from CampaignMember where id = :campm2.id].Status;
-    
-    system.assertEquals('Sent', campm1Status);
-    //We cannot test for "Sent -  Household duplicate", which should be the actual status that
-    //this campaign member gets, because that status is not visible without seealldata=true
-    system.assertEquals('Responded', campm2Status);  
-  
-  }
+    private static testMethod void shouldValidateDuplicateHouseholdCampaignMembers(){
+        UTIL_CustomSettingsFacade.getHouseholdsSettingsForTests(new npo02__Households_Settings__c (
+            npo02__Household_Rules__c = HH_Households.ALL_PROCESSOR,
+            npo02__Always_Rollup_to_Primary_Contact__c = false,
+            npo02__Enable_Opp_Rollup_Triggers__c = true,
+            npo02__Excluded_Account_Opp_Rectypes__c = null,
+            npo02__Excluded_Account_Opp_Types__c = null,
+            npo02__Excluded_Contact_Opp_Rectypes__c = null,
+            npo02__Excluded_Contact_Opp_Types__c = null,
+            npo02__Membership_Record_Types__c = null,
+            npo02__Use_Fiscal_Year_for_Rollups__c = true,
+            npo02__Rollup_N_Day_Value__c = 10,
+            npo02__Household_Mailing_List_ID__c = '0000000'
+        ));
+
+        // create test data
+        Account acct = new Account (
+            Name='Test Account'
+        );
+        insert acct;
+
+        Contact con = new Contact (
+            FirstName='Kevin',
+            LastName='Test',
+            AccountId = acct.Id,
+            npo02__TotalOppAmount__c = 100
+        );
+        insert con;
+
+        Id hhId = [SELECT npo02__Household__r.Id FROM Contact WHERE Id = :con.Id].npo02__Household__c;
+        System.assert(hhId != null);
+
+        Contact con2 = new Contact (
+            FirstName = 'Amy',
+            LastName = 'Test',
+            AccountId = acct.Id,
+            npo02__Household__c = hhId
+        );
+        insert con2;
+        System.assert(con2.npo02__Household__c != null);
```

Comment:
I believe this is testing the legacy household logic, so it would technically be possible (if something's not working) for that to be null. I didn't change this line though, other than capitalize the "S" in System.